### ISSUE TITLE
6.0.0 iOS: EXIF and GPS location does not work for Camera or Gallery Selected Images

### ISSUE THREAD

**Issue by codeconsole** [apache/cordova-plugin-camera#759]
This has been broken for years and either the documentation should be updated to not support it or a fix should be made. 

This camera plugin seems to work fine:
https://github.com/remoorejr/cordova-plugin-camera-with-exif
Consider referencing it for a solution on how to solve this issue.

But this is a core plugin and should be fixed.  Broken core plugins lead to a lot of people abandoning Cordova for other frameworks due to frustration.

I've created a sample app that demonstrates EXIF not working for Camera and Gallery.

https://github.com/codeconsole/exiftest

I've even created a pull request fix from other users contributions:

https://github.com/apache/cordova-plugin-camera/pull/712
This fix only fixes the Gallery and still results in the camera not showing Exif data.

I highly recommend at least merging the pull request if there is no desire left in the project to fix this plugin.  At least it will partially work.

**Comment by sc0ttdav3y** [comment]
Hey @codeconsole, thanks for your work here. 

I've also been juggling this issue for years, and I want to add some more context to this issue to respectfully ask if it can be looked at by those with the power to merge things.

The current environment:

1. Google Play no longer accepts Android apps without AAB builds, which are only offered with `cordova-android` 10.x
2. `cordova-android` now only supports Android X and above
3. The only previously working camera plugin ([cordova-plugin-camera-with-exif](https://github.com/remoorejr/cordova-plugin-camera-with-exif) is a fork of this, and it [does not yet support Android X](https://github.com/remoorejr/cordova-plugin-camera-with-exif/issues/56)
4. This plugin, which is the official one, claims to support exif data but it has not worked for years.
5. There have been a few good attempts by external developers to fix this over the years (and a big thanks to them!), but PRs remain un-merged or ignored.

I respect and want to thank the maintainers past and present. If the desire to fix this has gone or there's no bandwidth to do so, then just let us know. (@erisu, do you have any bandwidth to merge this PR)

As I see it, if there is no bandwidth to get proper exif support here, then one way forward may be in Javascript: i.e. capture the location data externally and then [add the exif data manually into the image in JS](https://github.com/hMatoba/piexifjs). But this won't help with exif on library photos, which this PR appears to solve.

**Comment by breautek** [comment]
> Google Play no longer accepts Android apps without AAB builds, which are only offered with cordova-android 10.x

Bundle support was added in cordova android 9.0. Recommend 9.1 for bug fixes related to bundles. You can use build.json file or --packageType=bundle to build aab bundles. You don't **need** 10.x to build bundles.

**Comment by erisu** [comment]
@codeconsole Can you explain how you are testing the PR changes?

When I took a photo using the changes from #712, the GPS Exif data does not exist.

JavaScript Sample:

```js
const cameraOptions = {
  quality: 100,
  destinationType: Camera.DestinationType.FILE_URI,
  sourceType: Camera.PictureSourceType.CAMERA,
  encodingType: Camera.EncodingType.JPEG,
  mediaType: Camera.MediaType.PICTURE,
  saveToPhotoAlbum: true,
}

cameraTakePhoto.onclick = function() {
  navigator.camera.getPicture(onPhotoDataSuccess, onFail, cameraOptions);
};
```

config.xml Sample
```xml
<platform name="ios">
    <preference name="CameraUsesGeolocation" value="true" />

    <edit-config target="NSCameraUsageDescription" file="*-Info.plist" mode="merge">
        <string>need camera access to take pictures</string>
    </edit-config>

    <edit-config target="NSPhotoLibraryUsageDescription" file="*-Info.plist" mode="merge">
        <string>need photo library access to get pictures from there</string>
    </edit-config>

    <edit-config target="NSLocationWhenInUseUsageDescription" file="*-Info.plist" mode="merge">
        <string>need location access to find things nearby</string>
    </edit-config>

    <edit-config target="NSPhotoLibraryAddUsageDescription" file="*-Info.plist" mode="merge">
        <string>need photo library access to save pictures there</string>
    </edit-config>
</platform>
```

From iOS's "Photo" app, I selected the photos from the gallery and airdropped them to my macOS. I opened the JPG photos with the "Previewer" app and opened the "Show Inspector" > "More Info" tab > "Exif" sub-tab. In this tab, there is no GPS location data.

From iOS's "Photo" app, I also tried checking by the "Places" category, and it's not there.

After taking the photo and clicking on the "Select" button, the location permission request dialog did open. I tried both options, "Allow Once" and "Allow While Using App" but neither of them made a difference.

There was also a minor issue that I noticed while testing. The location permission dialog will automatically close itself after a few seconds. This does not give people enough time to read and confirm. The dialog should not close until the user responds.

**Comment by codeconsole** [comment]
@erisu Thanks for the reply.

The pull request is the changes from https://github.com/apache/cordova-plugin-camera/pull/525 which is from 2 years ago, but without the Android changes since I believe the Android version was fixed. 

I assumed that @ryaa 's https://github.com/apache/cordova-plugin-camera/pull/525 never got merged and was neglected because it went out of sync. 

https://github.com/apache/cordova-plugin-camera/pull/712 brought 525 in sync, but does not fully address all the issues.  It is just @ryaa 's merged (iOS) branch which was the most working at the time.  Which was still better than noting and worked better than what the original provided.

Regarding the GSP information, I don't believe @ryaa 's  adequately addresses it, nor can I confirm it working.  In my opinion, it really just provides a solution for getting the exif info from the photos album.

**Comment by codeconsole** [comment]
@erisu I can tell you 100% that this pull request fixes exif information being saved when choosing from photo library.  Can you please merge it so we can at least have a somewhat working release?   This has been broken for over 2 years.  Wouldn't it be better to merge something that works rather than continuously releasing new versions of something broken?  Perhaps it would encourage more people in the community to get involved and fix the rest of the plugin rather than feeling ignored and then switching to other platforms such as capacitor.   At the very least, if you do not want the plugin to be fixed, update the documentation to state exif information is not supported and not available.

**Comment by erisu** [comment]
Resolved by #712
